AK/CircularDeque.h

```diff
@@ -35,6 +35,24 @@ namespace AK {
 template<typename T, size_t Capacity>
 class CircularDeque : public CircularQueue<T, Capacity> {
 public:
+    void enqueue_begin(T&& value)
+    {
+        const auto newHead = (this->m_head - 1 + Capacity) % Capacity;
+        auto& slot = this->elements()[newHead];
```

Comment:
No need for `this`

Reply:
Without `this`, this line won't compile.
```C++
const auto new_head = (this->m_head - 1 + Capacity) % Capacity;
```

Without `this`, this line also won't compile.
```C++
auto& slot = this->elements()[newHead];
```
[This FAQ](http://www.cs.technion.ac.il/users/yechiel/c++-faq/nondependent-name-lookup-members.html) explains it.